kore/src/Kore/Exec.hs

```diff
@@ -638,6 +655,52 @@ checkFunctions verifiedModule =
     checkResults eqns =
         mapM_ errorEquationRightFunction eqns $> ExitFailure 3
 
+{- | Returns true when both equations match the same term.  See:
+ - https://github.com/kframework/kore/issues/2472#issue-833143685
+-}
+bothMatch ::
+    MonadSimplify m =>
+    Equation VariableName ->
+    Equation VariableName ->
+    m Bool
+bothMatch eq1 eq2 =
+    let pre1 = Equation.requires eq1
+        pre2 = Equation.requires eq2
+        arg1 = fromMaybe Predicate.makeTruePredicate $ Equation.argument eq1
+        arg2 = fromMaybe Predicate.makeTruePredicate $ Equation.argument eq2
+        prio1 = fromMaybe Predicate.makeTruePredicate $ Equation.antiLeft eq1
+        prio2 = fromMaybe Predicate.makeTruePredicate $ Equation.antiLeft eq2
+        check =
+            Predicate.makeAndPredicate pre1 $
+                Predicate.makeAndPredicate pre2 $
+                    Predicate.makeAndPredicate arg1 $
+                        Predicate.makeAndPredicate arg2 $
+                            Predicate.makeAndPredicate prio1 prio2
+        check' = Predicate.mapVariables (pure mkConfigVariable) check
+     in (not . isBottom) <$> Predicate.simplify top check'
+
+{- | Checks if any function definition in the module carries two equations that both match
+ - same term.
+-}
+checkBothMatch ::
+    MonadSimplify m =>
+    -- | The main module
+    VerifiedModule StepperAttributes ->
+    m ExitCode
+checkBothMatch verifiedModule =
+    filterM (uncurry bothMatch) equations
+        >>= checkResults
+  where
+    equations = join $ map mkPairs $ Map.elems $ extractEquations verifiedModule
+    -- produces all 'in-order' pairs in a list
+    mkPairs xs =
```

Comment:
I remember adding some code used to generate unordered pairs (see the type's [smart constructor](https://github.com/kframework/kore/blob/98e241ede419e7a9e1266393104fcbd84fb796e4/kore/src/Kore/Internal/NormalizedAc.hs#L540)). I think we can use this type here to simplify things a little, let me know if this doesn't make sense for some reason.

Reply:
I'm not sure if that is relevant to this situation, by in-order I mean the order in which they are included in the list, so e.g. `[e0,e1,e2]` should yield `[(e0,e1), (e0,e2), (e1,e2)]` (although not necessarily in that order)

Reply:
From [Stackoverflow](https://stackoverflow.com/questions/34044366/how-to-extract-all-unique-pairs-of-a-list-in-haskell/34045121)
```
import Data.List

pairs :: [a] -> [(a, a)]
pairs l = [(x,y) | (x:ys) <- tails l, y <- ys]
```